processing/src/test/java/org/apache/druid/segment/join/JoinablesTest.java

```diff
@@ -190,6 +196,176 @@ public boolean isDirectlyJoinable(DataSource dataSource)
     Assert.assertNotSame(Function.identity(), segmentMapFn);
   }
 
+  @Test(expected = IAE.class)
+  public void test_computeJoinDataSourceCacheKey_noClauses()
+  {
+    DataSourceAnalysis analysis = EasyMock.mock(DataSourceAnalysis.class);
+    EasyMock.expect(analysis.getPreJoinableClauses()).andReturn(Collections.emptyList()).anyTimes();
+    EasyMock.replay(analysis);
+    Joinables joinables = new Joinables(new JoinableFactoryWithCacheKey());
+    joinables.computeJoinDataSourceCacheKey(analysis);
+  }
+
+  @Test
+  public void test_computeJoinDataSourceCacheKey_noHashJoin()
+  {
+
+    PreJoinableClause clause_1 = makeGlobalPreJoinableClause("dataSource_1", "x == \"j.x\"", "j.");
```

Comment:
We don't use underscore in variable names. Please rename all variables containing underscores.